### ISSUE TITLE
Cannot open .image.so

### ISSUE THREAD

**Issue by Xevib** [DTVD/rainbowstream#142]
I'm runing rainbowstream at first time and i get the following error

Traceback (most recent call last):
  File "/usr/bin/rainbowstream", line 9, in <module>
    load_entry_point('rainbowstream==1.3.1', 'console_scripts', 'rainbowstream')()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 565, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2704, in load_entry_point
    return ep.load()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2377, in load
    return self.resolve()
  File "/usr/lib/python2.7/site-packages/pkg_resources/__init__.py", line 2383, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "build/bdist.linux-x86_64/egg/rainbowstream/rainbow.py", line 22, in <module>
  File "build/bdist.linux-x86_64/egg/rainbowstream/draw.py", line 15, in <module>
  File "build/bdist.linux-x86_64/egg/rainbowstream/c_image.py", line 25, in <module>
  File "build/bdist.linux-x86_64/egg/rainbowstream/c_image.py", line 21, in call_c
  File "/usr/lib/python2.7/ctypes/__init__.py", line 443, in LoadLibrary
    return self._dlltype(name)
  File "/usr/lib/python2.7/ctypes/__init__.py", line 365, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: /home/xevi/.image.so: cannot open shared object file: No such file or directory

I'm using an Arch linux with Python3.5

**Comment by alfrex** [comment]
On 29/11/15 03:24, Xevi wrote:
> I'm runing rainbowstream at first time and i get the following error
> 
> Traceback (most recent call last):
> File "/usr/bin/rainbowstream", line 9, in
> load_entry_point('rainbowstream==1.3.1', 'console_scripts',
> 'rainbowstream')()
> File "/usr/lib/python2.7/site-packages/pkg_resources/*init*.py", line
> 565, in load_entry_point
> return get_distribution(dist).load_entry_point(group, name)
> File "/usr/lib/python2.7/site-packages/pkg_resources/*init*.py", line
> 2704, in load_entry_point
> return ep.load()
> File "/usr/lib/python2.7/site-packages/pkg_resources/*init*.py", line
> 2377, in load
> return self.resolve()
> File "/usr/lib/python2.7/site-packages/pkg_resources/*init*.py", line
> 2383, in resolve
> module = *import*(self.module_name, fromlist=['*name*'], level=0)
> File "build/bdist.linux-x86_64/egg/rainbowstream/rainbow.py", line 22, in
> File "build/bdist.linux-x86_64/egg/rainbowstream/draw.py", line 15, in
> File "build/bdist.linux-x86_64/egg/rainbowstream/c_image.py", line 25, in
> File "build/bdist.linux-x86_64/egg/rainbowstream/c_image.py", line 21,
> in call_c
> File "/usr/lib/python2.7/ctypes/*init*.py", line 443, in LoadLibrary
> return self./dlltype(name)
> File "/usr/lib/python2.7/ctypes/__init_/.py", line 365, in *init*
> self._handle = _dlopen(self._name, mode)
> OSError: /home/xevi/.image.so: cannot open shared object file: No such
> file or directory
> 
> I'm using an Arch linux with Python3.5

Hi, .image.so is a hidden file in home directory created while run
rainbowstream.

I'am using rainbowstream from AUR, but on Manjaro, Precisely it is
running on python 3.5, so you can try uninstall and compile against
packages from python 3, for example python-setuptools instead
python2-setuptools.

Check your rainbowstream what version of python is calling:

nano /usr/bin/rainbowstream

#!/usr/bin/python

or

#!/usr/bin/python2

if is version 2, try deleting the number 2, so:

#!/usr/bin/python

and run again rainbowstream

Another option is to check if all dependencies are installed.

Greetings.

**Comment by Xevib** [comment]
Sorry I checked the /usr/bin/rainbowstream and it was with the python2 shebang, I reinstalled it with python3 and and I have the following message
Traceback (most recent call last):
  File "/usr/bin/rainbowstream", line 9, in <module>
    load_entry_point('rainbowstream==1.3.1', 'console_scripts', 'rainbowstream')()
  File "/usr/lib/python3.5/site-packages/pkg_resources/__init__.py", line 565, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2704, in load_entry_point
    return ep.load()
  File "/usr/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2377, in load
    return self.resolve()
  File "/usr/lib/python3.5/site-packages/pkg_resources/__init__.py", line 2383, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/usr/lib/python3.5/site-packages/rainbowstream/rainbow.py", line 25, in <module>
    from .consumer import *
ImportError: No module named 'rainbowstream.consumer'

I installed rainbowstream with pip from the git repo

Thank you

**Comment by alfrex** [comment]
Using python2, python2-pillow (or python2-pil) manage the image's issues.

Greetings.

**Comment by DTVD** [comment]
HI @Xevib . Need to specify that RainbowStream **only** can be installed via `pip` but not directly from this repo. This repo does not contains Twitter API key information so you got that "No module named rainbowstream.consumer". 

I see you have all Python 2 and 3. You should pick only one and try *The quick way* or *The recommend way* in README.md

**Comment by Xevib** [comment]
Thank you I installed it using pip in python3 version and it worked

**Comment by cavkirov** [comment]
I've always installed rainbowstream the quick way but now I have the same issue as outlined here about the .image.so file. It was running under python2.7, so I've now got it installed under python3.5, but still same thing. Not sure what else I should do?

**Comment by cavkirov** [comment]
Just installed the recommended way and got the same error.

**Comment by cavkirov** [comment]
Ended up copying the .image.so file from another computer and it worked/

**Comment by ransur0t** [comment]
I experienced the same issue with the missing ~/.image.so file on an install using the recommended way on a system running Debian stretch.  Per @cavkirov, I copied the file from another pc with a functional rainbow stream installation to solve to problem.

**Comment by woohgit** [comment]
The problem here is the missing `C compiler` on those systems.

rainbowstream would compile it automatically, but doesn't handle the missing C compiler. It does not give you any error. What it does is actually:

```shell
$ cc -fPIC -shared -o ~/.image.so ./lib/python3.5/site-packages/rainbowstream/image.c
```
So all you need to do is install the c compiler and re-run `rainbowstream` and voila. On ubuntu you can do it via:
 ```
sudo apt-get install build-essentials
```

**Comment by OrangeTsar** [comment]
Ran into this issue on gallium and solus,  @alfrex 's first response took care of it on gallium, @woohgit pointed me in the direction for solus, where the command to get the C compiler is 
`sudo eopkg install -c system.devel
` and then reinstalling with pip. No need to copy files from other computers.

**Comment by ptytb** [comment]
@OrangeTsar please consider testing [my pull request](https://github.com/orakaro/rainbowstream/pull/256), which closes #142 #246
